libgdf/src/transpose.cu

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2018 NVIDIA Corporation.  All rights reserved. */
+
+#include "gdf_table.cuh"
+#include <gdf/gdf.h>
+#include <thrust/for_each.h>
+#include <thrust/iterator/counting_iterator.h>
+#include <memory>
+#include <stdio.h>
+
+gdf_error gdf_transpose(size_t ncols,
+                        gdf_column** in_cols,
+                        gdf_column** out_cols)
```

Comment:
@kkraus14 @randerzander @dantegd discussed that, at least for now, we're going to limit transpose to supporting only columns of all the same type. I would add a check to make sure that this is the case, and return a column dtype mismatch error if not. (All the input and output columns should be the same type).